apis/python/src/tiledbsoma/util.py

```diff
@@ -191,3 +191,50 @@ def dense_index_to_shape(
         return stop - start + 1
 
     raise TypeError("coordinates must be tuple of int or slice")
+
+
+def check_type(
+    name: str,
+    actual_type: Type[Any],
+    expected_types: Tuple[Type[Any], ...],
+) -> None:
+    """
+    Keystroke-saver for runtime type-checking.
+
+    Context: while in principle we could use this to type-check every argument to every
+    function/method, in practice the most necessary guardrail to offer to the user is around schema
+    and container types for I/O. Specfically, SOMA's type system is Arrow while computational
+    biologists are often using SOMA to store results of Scanpy analyses. It's extra-important that
+    if anyone naïvely attempts to write a Pandas DataFrame, SciPy CSR matrix, etc. to a SOMA object,
+    they should get clear and actionable insight on that.
+
+    Parameters
+    ----------
+    name: str
+        Name of the argument in question, as used by the caller.
+
+    XXX CHANGE ME
+    actual_type: Any
+        Value of the argument. Note that it would be much easier for us to pass the _type_ of the
+        value rather than the full value. Unfortunately, we cannot: within the SOMA API we have
+        argument names like `type`. Then the caller would have to pass us `type(type)`. But this is
+        a name clash between Python's `type` as a function name, and SOMA's `type` as an argument
+        name.
+
+    expected_type: Any
+        Things like `pyarrow.Tensor`, etc.
+
+    Returns
+    -------
+    None. Raises `TypeError` on mismatch.
```

Comment:
I would make the first line of this describe its function more precisely:

`Verifies the type of an argument or produces a useful error message.`

I also think that everything including and after `Parameters` can be deleted—the names are clear, the function is internal-only, and the implementation is a number of lines you can count on one hand.

(also consider checking margins so things wrap around 88 chars to match black’s formatting of the rest of the code)

Reply:
Sorry this is not ready for review -- this entire doc block with `XXX CHANGE ME` needs updating.

I should have left this in draft, sorry.